code/datums/components/bouquet.dm

```diff
@@ -0,0 +1,88 @@
+// WARNING: attach this component only to things that have can_bouquet as a var.
+// so far this includes /obj/item/clothing/head/flower and /obj/item/plant
+// Actual bouquets are in 'code/obj/item/bouquet.dm'
+
+/* STEPS TO ADDING A NEW FLOWER TO BE BOUQUETABLE
+1.	create three sprites for them in bouquets.dmi. names them name_l, name_m, and name_r for left middle and right.
+	Visually, the middle ones cover the ones at the sides. Keep that in mind.
+2.	Go to the flower involved and add the component in the New() proc. Look at lavender to see how it's done.
+3. Optionally, at the bottom of code/obj/item/bouquet.dm, create a subtype of premade monotype bouquets
+*/
+
+TYPEINFO(/datum/component/bouquet)
+	initialization_args = list()
+
+/// the bouquet component, that allows flowers of various parentage to be wrapped into bouquets
+/datum/component/bouquet
+
+/datum/component/bouquet/Initialize()
+	. = ..()
+	if(!istype(parent, /obj/item))
+		return COMPONENT_INCOMPATIBLE
+	var/obj/item/W = parent
+	if (!W.is_valid_icon_state("[W.icon_state]_l", 'icons/obj/items/bouquets.dmi'))
+		return COMPONENT_INCOMPATIBLE // it doesn't a matching left icon
+	if (!W.is_valid_icon_state("[W.icon_state]_m", 'icons/obj/items/bouquets.dmi'))
+		return COMPONENT_INCOMPATIBLE // it doesn't a matching middle icon
+	if (!W.is_valid_icon_state("[W.icon_state]_r", 'icons/obj/items/bouquets.dmi'))
+		return COMPONENT_INCOMPATIBLE // it doesn't a matching right icon
+	RegisterSignal(parent, COMSIG_ATTACKBY, .proc/attackby)
+	RegisterSignal(parent, COMSIG_ITEM_AFTERATTACK, .proc/attack)
+
+/datum/component/bouquet/UnregisterFromParent()
+	UnregisterSignal(parent, COMSIG_ATTACKBY)
+	UnregisterSignal(parent, COMSIG_ITEM_AFTERATTACK)
+	. = ..()
+
+/datum/component/bouquet/proc/attack(obj/item/source, atom/W, mob/user, reach, params)
+	// if we attack a bouquet with a flower, add it.
+	if (!istype(W, /obj/item/bouquet))
+		return
+	var/obj/item/bouquet/B = W
+	if (B.flowernum >= B.max_flowers)
+		boutput(user, "This bouquet is full!")
+		return
+	B.add_flower(source, user)
+	B.notaddedflower = TRUE
```

Comment:
Set truthy return value and it will override typical calls. I think that should resolve this work around.

https://github.com/goonstation/goonstation/blob/5a1658e6b8321becace9dd24f259be9988170610/code/atom.dm#L829C16-L829C16

Reply:
Okay, well I've put it in. Hopefully it works.